common/src/cellular_common.c

```diff
@@ -1127,3 +1121,27 @@ CellularError_t _Cellular_LibCleanup( CellularHandle_t cellularHandle )
 }
 
 /*-----------------------------------------------------------*/
+
+/* This function is provided as common code to cellular module porting.
+ * Vendor may choose to use this function or use their implementation. */
+/* coverity[misra_c_2012_rule_8_7_violation]. */
+CellularPktStatus_t _Cellular_AtcmdRequestWithCallback( CellularContext_t * pContext,
+                                                        CellularAtReq_t atReq )
+{
+    /* Parameters are checked in this function. */
+    return _Cellular_TimeoutAtcmdRequestWithCallback( pContext, atReq, PACKET_REQ_TIMEOUT_MS );
+}
+
+/*-----------------------------------------------------------*/
+
+/* This function is provided as common code to cellular module porting.
+ * Vendor may choose to use this function or use their implementation. */
+/* coverity[misra_c_2012_rule_8_7_violation]. */
+CellularPktStatus_t _Cellular_TimeoutAtcmdRequestWithCallback( CellularContext_t * pContext,
+                                                               CellularAtReq_t atReq,
+                                                               uint32_t timeoutMS )
+{
+    return _Cellular_PktHandler_TimeoutAtcmdRequestWithCallback( pContext, atReq, timeoutMS );
```

Comment:
There are two reasons I move these APIs from cellular_pkthander.c to cellular_common.c:
1. _Cellular_AtcmdRequestWithCallback and _Cellular_TimeoutAtcmdRequestWithCallback are declared in cellular_common.h, but implemented in cellular_pkthandler.c, I think that doesn't make sense.
2. When I do unit test on cellular_3gpp_api.c, what I need to include is cellular_common.c and ... cellular_pkthandler.c, for this purpose, I need to implement some platform APIs such as xQueseSend, xQueseCreate used by cellular_pkthanlder.c, I even can't mock these APIs because they are using the header file.
3. And you are right, those other public APIs implemented in cellular_pkthandler.c but declared in cellular_common.c should be do the same wrap as what I do for _Cellular_PktHandler_TimeoutAtcmdRequestWithCallback, I think this changes could be done in next PR, because this is also regarding to the unit test coverage.

Reply:
I agree with point 3. Let's move the internal API changes in a separate PR.